### PR TITLE
Skip release check sometimes

### PR DESCRIPTION
**Pull request by jkimbo**
## Description

This PR updates the release check workflow to skip checks if it's a PR from dependabot or no files in the main library have changed.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

*

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by patrick91** [comment]
@jkimbo have you seen this? https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestpaths

**Comment by jkimbo** [comment]
> @jkimbo have you seen this? https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestpaths

Oh yeah that makes much more sense 🤦

**Review by patrick91 — APPROVED** [review]

**Comment by patrick91** [comment]
Going to do some changes to this PR now that we have also changed the action 😊

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/), so make sure to pick the appropriate type. If in doubt feel free to ask :)

**Comment by patrick91** [comment]
Tested here :) https://github.com/strawberry-graphql/strawberry/pull/1143

**Review by patrick91 — APPROVED** [review]

File: .github/workflows/release-check.yml [hunk 1]
```diff
@@ -4,7 +4,8 @@ on:
   pull_request_target:
     types: [synchronize, reopened, opened, ready_for_review]
     branches:
-    - main
+    # TODO: revert
```
Comment (BryceBeagle) [inline .github/workflows/release-check.yml]: :thinking:

**Pull request merged**